### DISCUSSION TITLE
How to scale in stable pods

### DISCUSSION THREAD

**Discussion by lcc3108** [argoproj/argo-rollouts#3952] [Q&A]
Hello

I want to use argo rollout for servers with hundreds of thousands of WebSocket connections.

The scenario is as follows.

1. 100 bluepods receive traffic
2. Create 100 green pods with new version deployment (traffic switching X)
3. 100% Traffic switching to green pods
4. Keep the connection with blue pods because it's a web socket
5. Shut down some bluepods to migrate the bluepods connection to green
6. Repeat step 5 to all shutdown blue pods

setStableScale was not added by the https://github.com/argoproj/argo-rollouts/issues/1029 issue, but is there a way to use it similarly?

The key is to scale down bluepods regardless of traffic while maintaining the number of green pods.

**Comment by jsirianni** [comment]
I was hoping this behavior was available as well. I do not want to immediately cut over 100% of my websocket connections. Ideally, the blue replicaset would scale down one at a time instead of all at once. This causes a massive re-connection surge.
